### PR TITLE
[test-api-update] integration_tests

### PR DESCRIPTION
**Pull request by smackesey**
## Summary & Motivation

Convert all integration test suites fully over to job/op/graph APIs.

## How I Tested These Changes

BK

### CODE REVIEW

**Comment by smackesey** [comment]
Current dependencies on/for this PR:
* master
  * **PR #13145** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13145" target="_blank">Graphite</a> 
    * **PR #13108** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13108" target="_blank">Graphite</a> 
      * **PR #13168** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13168" target="_blank">Graphite</a> 
        * **PR #13169** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13169" target="_blank">Graphite</a> 
          * **PR #13081** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13081" target="_blank">Graphite</a> 
            * **PR #13083** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13083" target="_blank">Graphite</a> 
            * **PR #12770** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12770" target="_blank">Graphite</a> 
              * **PR #12769** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12769" target="_blank">Graphite</a> 
              * **PR #13170** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13170" target="_blank">Graphite</a> 
                * **PR #13171** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13171" target="_blank">Graphite</a> 
                  * **PR #13188** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13188" target="_blank">Graphite</a> 
                    * **PR #13172** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13172" target="_blank">Graphite</a> 
                      * **PR #13192** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13192" target="_blank">Graphite</a> 
                        * **PR #13301** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13301" target="_blank">Graphite</a> 
                        * **PR #13295** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13295" target="_blank">Graphite</a> 
                          * **PR #13296** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13296" target="_blank">Graphite</a> 
                            * **PR #13294** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13294" target="_blank">Graphite</a> 
                              * **PR #13297** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13297" target="_blank">Graphite</a> 
                                * **PR #13303** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13303" target="_blank">Graphite</a> 
                                  * **PR #13299** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13299" target="_blank">Graphite</a> 
                                    * **PR #13298** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13298" target="_blank">Graphite</a> 
                                      * **PR #13302** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13302" target="_blank">Graphite</a> 
                                        * **PR #13311** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13311" target="_blank">Graphite</a> 
                                          * **PR #13312** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13312" target="_blank">Graphite</a> 
                                            * **PR #13316** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13316" target="_blank">Graphite</a> 
                                              * **PR #13317** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13317" target="_blank">Graphite</a> 
                                                * **PR #13318** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13318" target="_blank">Graphite</a> 
                                                  * **PR #13321** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13321" target="_blank">Graphite</a>  👈
                                                    * **PR #13320** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13320" target="_blank">Graphite</a> 
                      * **PR #13173** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13173" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/13321?utm_source=stack-comment).

**Comment by smackesey** [comment]
Current dependencies on/for this PR:
* master
  * **PR #13145** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13145" target="_blank">Graphite</a> 
    * **PR #13108** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13108" target="_blank">Graphite</a> 
      * **PR #13168** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13168" target="_blank">Graphite</a> 
        * **PR #13169** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13169" target="_blank">Graphite</a> 
          * **PR #13081** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13081" target="_blank">Graphite</a> 
            * **PR #13083** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13083" target="_blank">Graphite</a> 
            * **PR #12770** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12770" target="_blank">Graphite</a> 
              * **PR #12769** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12769" target="_blank">Graphite</a> 
              * **PR #13170** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13170" target="_blank">Graphite</a> 
                * **PR #13171** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13171" target="_blank">Graphite</a> 
                  * **PR #13188** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13188" target="_blank">Graphite</a> 
                    * **PR #13172** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13172" target="_blank">Graphite</a> 
                      * **PR #13192** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13192" target="_blank">Graphite</a> 
                        * **PR #13301** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13301" target="_blank">Graphite</a> 
                        * **PR #13295** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13295" target="_blank">Graphite</a> 
                          * **PR #13296** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13296" target="_blank">Graphite</a> 
                            * **PR #13294** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13294" target="_blank">Graphite</a> 
                              * **PR #13297** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13297" target="_blank">Graphite</a> 
                                * **PR #13303** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13303" target="_blank">Graphite</a> 
                                  * **PR #13299** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13299" target="_blank">Graphite</a> 
                                    * **PR #13298** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13298" target="_blank">Graphite</a> 
                                      * **PR #13302** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13302" target="_blank">Graphite</a> 
                                        * **PR #13311** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13311" target="_blank">Graphite</a> 
                                          * **PR #13312** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13312" target="_blank">Graphite</a> 
                                            * **PR #13316** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13316" target="_blank">Graphite</a> 
                                              * **PR #13317** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13317" target="_blank">Graphite</a> 
                                                * **PR #13318** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13318" target="_blank">Graphite</a> 
                                                  * **PR #13321** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13321" target="_blank">Graphite</a>  👈
                                                    * **PR #13320** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13320" target="_blank">Graphite</a> 
                      * **PR #13173** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13173" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/13321?utm_source=stack-comment).

File: integration_tests/test_suites/backcompat-test-suite/dagit_service/legacy_repo.py [hunk 1]
```diff
@@ -1,4 +1,7 @@
-# pylint: skip-file
+# type: ignore
+
+# This file is only loaded by old versions of dagster during backcompat testing.
+
```
Comment (dpeng817) [inline integration_tests/test_suites/backcompat-test-suite/dagit_service/legacy_repo.py:4]: thanks for adding

File: integration_tests/test_suites/daemon-test-suite/repo.py [hunk 2]
```diff
@@ -1,40 +1,39 @@
-from dagster import RunRequest, op, repository, schedule, sensor
-from dagster._legacy import pipeline
+from dagster import RunRequest, job, op, repository, schedule, sensor
 
 
 @op()
```
Comment (dpeng817) [inline integration_tests/test_suites/daemon-test-suite/repo.py:4]: mind deleting that paren

File: integration_tests/test_suites/daemon-test-suite/repo.py [hunk 3]
```diff
@@ -1,40 +1,39 @@
...
-def other_foo_pipeline():
-    foo_solid()
+@job
+def other_foo_job():
+    foo_op()
 
 
 @schedule(
-    job_name="foo_pipeline",
+    job_name="foo_job",
```
Comment (dpeng817) [inline integration_tests/test_suites/daemon-test-suite/repo.py:20]: fwiw you can just provide the job directly now

**Review by dpeng817 — APPROVED** [review]

**Pull request merged**